mllib/src/main/scala/org/apache/spark/mllib/tree/model/treeEnsembleModels.scala

```diff
@@ -166,6 +156,56 @@ class GradientBoostedTreesModel(
 
 object GradientBoostedTreesModel extends Loader[GradientBoostedTreesModel] {
 
+  /**
+   * Method to compute initial error and prediction as a RDD for the first
+   * iteration of gradient boosting.
+   * @param data RDD of [[org.apache.spark.mllib.regression.LabeledPoint]]
+   * @param initTreeWeight: learning rate assigned to the first tree.
+   * @param initTree: first DecisionTreeModel
+   * @param loss: evaluation metric
+   * @return a RDD with each element being a zip of the prediction and error
+   *         corresponding to every sample.
+   */
+  def computeInitialPredictionAndError(
+      data: RDD[LabeledPoint],
+      initTreeWeight: Double,
+      initTree: DecisionTreeModel, loss: Loss): RDD[(Double, Double)] = {
+    data.map { i =>
+      val pred = initTreeWeight * initTree.predict(i.features)
+      val error = loss.computeError(pred, i.label)
+      (pred, error)
+    }
+  }
+
+  /**
+   * Method to update a zipped predictionError RDD
+   * (as obtained with computeInitialPredictionAndError)
+   * @param data RDD of [[org.apache.spark.mllib.regression.LabeledPoint]]
+   * @param predictionAndError: predictionError RDD
+   * @param currentTreeWeight: learning rate.
+   * @param currentTree: first DecisionTree
+   * @param loss: evaluation metric
+   * @return a RDD with each element being a zip of the prediction and error
+   *         corresponing to each sample.
+   */
+  def updatePredictionError(
+    data: RDD[LabeledPoint],
+    predictionAndError: RDD[(Double, Double)],
+    currentTreeWeight: Double,
+    currentTree: DecisionTreeModel,
+    loss: Loss): RDD[(Double, Double)] = {
+
+    data.zip(predictionAndError).mapPartitions { iter =>
+      iter.map {
+        case (point, (pred, error)) => {
```

Comment:
"point" -->  (Use the same variable name for a labeled point everywhere for consistency.  I tend to use "lp")